datafusion/src/physical_plan/repartition.rs

```diff
@@ -723,4 +743,105 @@ mod tests {
 
         assert_batches_sorted_eq!(&expected, &batches);
     }
+
+    #[tokio::test]
```

Comment:
I did not write a test that early shutdown stop consuming input (because repartition exec uses unbounded channels, it buffers indefinitely and thus the timing control has to be very precise)

🤔  mostly that sounds like an excuse